lib/imgix/client.rb

```diff
@@ -60,6 +52,27 @@ def prefix
 
     private
 
+    def create_request(endpoint, img_url)
+      req = Net::HTTP::Post.new(endpoint.path)
+      req["Content-Type"] = "application/json"
+      req["Authorization"] = "Bearer #{@api_key}"
+      req["User-Agent"] = "imgix #{@library}-#{@version}"
+      req.body = json_data(img_url)
+
+      req
+    end
```

Comment:
No change needed, but I think this abstraction could be improved slightly. I think it'd be better to have a method `def create_request(endpoint, json_data)` so that this method isn't coupled to the purge JSON data. Then you would call this method with `create_request(endpoint, purge_json_data(url))`

Reply:
Yeah, originally, I wrote this PR a lot differently. In that other incarnation, we'd be able to support multiple endpoint and body/data payloads. I wonder if it would be worth it to write this like:

```rb
def create_request(endpoint, resource, f)
  # -- snip --
  req.body = f(resource)
``` 

Where calls to this could look like:
```rb
create_request(endpoint, url, v1_data)
create_request(endpoint, url, v2_data)
create_request(endpoint, json_obj, json_to_v1)
```

I like `create_request(endpoint, purge_json_data(url))`, but it's a little too implicit (semantically) IMO. But this great `create_request(2)` can benefit from becoming `create_request(3)` which seems to decouple things well while also being explicit about the decoupling of things.

For example, `create_request` takes an `endpoint`, `resource`, and `f`ormatter function that produces the desired data/message-body.

Yeah, I like this; if we end up supporting v1 and v2 requests (post `imgix-rb 4.0.0`) then this decoupling is a nice win :)

Reply:
Good discussion here - can you elaborate a little bit on what this meant: "but it's a little too implicit (semantically) IMO"?

I'm not too sure why `create_request` needs an extra parameter/knowledge of the resource since it doesn't explicitly use it - seems like a smell to me.

End of the day this doesn't really matter, so don't worry too much about it. I think the best course of action is for you just to choose what signature you like the best and go with that, since you have the best context in this issue right now.